worlds/shivers/__init__.py

```diff
@@ -65,153 +71,184 @@ def create_regions(self) -> None:
+            storage_items += [self.create_item(name) for name, data in item_table.items() if
+                              data.type == ItemType.POT_DUPLICATE]
         elif self.options.full_pots == "complete":
-            storageitems += [self.create_item(name) for name, data in item_table.items() if data.type == 'potduplicate_type2']
-            storageitems += [self.create_item("Empty") for i in range(10)]
+            storage_items += [self.create_item(name) for name, data in item_table.items() if
+                              data.type == ItemType.POT_COMPELTE_DUPLICATE]
+            storage_items += [self.create_item("Empty") for _ in range(10)]
         else:
+            pieces = [self.create_item(name) for name, data in item_table.items() if
+                      data.type == ItemType.POT_DUPLICATE]
+            complete = [self.create_item(name) for name, data in item_table.items() if
+                        data.type == ItemType.POT_COMPELTE_DUPLICATE]
             for i in range(10):
-                #Pieces
+                # Pieces
                 if self.pot_completed_list[i] == 0:
-                    storageitems += [self.create_item(self.item_id_to_name[self.shivers_item_id_offset + 70 + i])]
-                    storageitems += [self.create_item(self.item_id_to_name[self.shivers_item_id_offset + 80 + i])]
-                #Complete
+                    storage_items.append(pieces[i])
+                    storage_items.append(pieces[i + 10])
+                # Complete
                 else:
-                    storageitems += [self.create_item(self.item_id_to_name[self.shivers_item_id_offset + 140 + i])]
-                    storageitems += [self.create_item("Empty")]
+                    storage_items.append(complete[i])
+                    storage_items.append(self.create_item("Empty"))
 
-        storageitems += [self.create_item("Empty") for i in range(3)]
+        storage_items += [self.create_item("Empty") for _ in range(3)]
 
         state = self.multiworld.get_all_state(True)
 
-        self.random.shuffle(storagelocs)
-        self.random.shuffle(storageitems)
-        
-        fill_restrictive(self.multiworld, state, storagelocs.copy(), storageitems, True, True)
+        self.random.shuffle(storage_locs)
+        self.random.shuffle(storage_items)
 
-        self.storage_placements = {location.name: location.item.name for location in storagelocs}
+        fill_restrictive(self.multiworld, state, storage_locs.copy(), storage_items, True, True)
 
-    set_rules = set_rules
+        self.storage_placements = {location.name.replace("Storage: ", ""): location.item.name.replace(" DUPE", "") for
+                                   location in storage_locs}
 
-    def fill_slot_data(self) -> dict:
+        for name, data in item_table.items():
+            if data.type == ItemType.GOAL:
+                goal = self.create_item(name)
+                self.get_location("Mystery Solved").place_locked_item(goal)
```

Comment:
You should be doing this when you create the location/item. `pre_fill` can run into problems with plando